### ISSUE TITLE
Seems like I can't use object as a parameter.

### ISSUE THREAD

**Issue by kegor** [Starcounter/Starcounter.Linq#52]
I tried to use query like (to get all objects except one):
```csharp
var products =
DbLinq.Objects<Product>().Where(x => x.ProductVariantGroup == productPage.Data.ProductVariantGroup && x != productPage.Data);
```
And I'm getting this exception:
![image](https://user-images.githubusercontent.com/17431807/43826963-4e46635e-9b01-11e8-850c-eac8ee58698b.png)

**Comment by un-tone** [comment]
This case is also good to be fixed:
```cs
DbLinq.Objects<Product>().Where(product => !assortmentsInStore.Contains(product))
```

**Comment by un-tone** [comment]
The investigation showed that the reason is not in "object as a parameter", but in direct using predicate parameters. For example:

```cs
// it's OK
Objects<Department>().Where(d => d.Company == company);
```

```cs
// here the bug
Objects<Department>().Where(d => d == department);
```

**Comment by un-tone** [comment]
Fixed in versions [1.1](https://github.com/Starcounter/Starcounter.Linq/releases/tag/1.1.0), [2.1](https://github.com/Starcounter/Starcounter.Linq/releases/tag/2.1.0), [3.1](https://github.com/Starcounter/Starcounter.Linq/releases/tag/3.1.0)
